### ISSUE TITLE
[enhancement Graph Construction] reinstantiate `FeatTransform` objects at inference time 

### ISSUE THREAD

**Issue by milianru** [awslabs/graphstorm#1067]
Hey,

it would be helpful to setup comparable transformers as were used during graph construction for the model operation in an inductive realtime setting. 

Currently the `process_data()` main function allows easy processing of bulk data, covering all required steps, but also abstracts and covers the state of the transformers (`FeatTransform`s). To easily operate the ML model, I would like to be enabled to reinstantiate the transformers. 

As an additional output of `process_data` it would be great to dump the transform objects after calling the `process_node_data` and `process_edge_data`? I guess there different feasible options such as:
-  `.pkl` like it is common with scikit-learn transformers
-  or a JSON file containing metadata for restoring the state of the `FeatTransform`

At the moment I am starting with rather simple, numeric transformations. For these transformations I want to gather relevant metadata (min, max, ...) and reimplement a similar processing logic for the realtime application. 

Are there any other suggestions or perspectives on this topic?

Thanks in advance.

**Comment by classicsong** [comment]
Hi. This information is automatically saved into a JSON file. If you are using gconstruct, you can use `--output-conf-file` to define the filename of the output JSON file which will store the transformation meta-data.

**Comment by zhjwy9343** [comment]
Hi @milianru, if you use gconstruct.construct_graph.py command, by default, the command will create a file called `data_transform_new.json`, which includes the transformation details.

May you try out the `Use Your Own Data` tutorial, https://graphstorm.readthedocs.io/en/latest/tutorials/own-data.html#use-your-own-data, and run the gconstruct command, you will find this new JSON file.

**Comment by milianru** [comment]
Hi @classicsong  and @zhjwy9343,

thanks for directing me to `--output-conf-file` and `data_transform_new.json`. This clarifications will definitely help and get me started. 

I guess we didn't see it before, for two reasons. (a) We did define min and max values in the for the `NumericalMinMaxTransform` and some parameters for the `RankGaussTransform`.  (b) We also called it `config.json`.  
In the end the created file via `--output-conf-file` looked pretty much like the input construction configuration file. If i overlooked it correctly, no new transform metadata was added.

For our numerical transformations we will be using the transformers such as the `NumericalMinMaxTransform`. The `RankGaussTransform` doesn't seem to be applicable for us, since it is supposed to perform the transformation on the full feature set. In our case, we need to process only small subgraphs at a time, so the 'RankGaussTransform` won't work out of the box. But that is fine.
